### PR TITLE
[thermalctld] Update threshold values

### PR DESCRIPTION
**Pull request by d-dashkov**
Signed-off-by: d-dashkov <Dmytro_Dashkov@Jabil.com>

#### Description
The thermal daemon has been modified, before setting the thresholds values in Redis, compares them from Redis with the platform API values, if they differ, the update process sets these new values for the platform.

#### Motivation and Context
Associated with PR: https://github.com/Azure/sonic-platform-common/pull/180

### CODE REVIEW

**Review by jleveque — CHANGES_REQUESTED** [review]
Please fix unit test failures

File: sonic-thermalctld/scripts/thermalctld [hunk 1]
```diff
@@ -640,6 +640,36 @@ class TemperatureUpdater(logger.Logger):
                                         )
         warning = warning | temperature_status.under_temperature
 
+        THRESH_NAME_LIST = ['high_threshold', 'low_threshold', 'critical_high_threshold', 'critical_low_threshold']
```
Comment (Junchao-Mellanox) [inline sonic-thermalctld/scripts/thermalctld:643]: The following 3 variables can be class level

File: sonic-thermalctld/scripts/thermalctld [hunk 2]
```diff
@@ -640,6 +640,36 @@ class TemperatureUpdater(logger.Logger):
...
+            if second == NOT_AVAILABLE:
+                continue
+
+            th_type = 0
+            if first in THRESH_NAME_LIST:
+                th_type = THRESH_NAME_LIST.index(first)
+            else:
+                continue
+
+            if second != str(THRESH_COMP_VAL_LIST[th_type]):
```
Comment (Junchao-Mellanox) [inline sonic-thermalctld/scripts/thermalctld:659]: "second" here is the value of a threshold, while THRESH_COMP_VAL_LIST[th_type] is a element in THRESH_COMP_VAL_LIST, i suppose they never equal.

File: sonic-thermalctld/scripts/thermalctld [hunk 3]
```diff
@@ -640,6 +640,36 @@ class TemperatureUpdater(logger.Logger):
...
+                try:
+                    if THRESH_FUNC_LIST[th_type](float(second)) is True:
```
Comment (Junchao-Mellanox) [inline sonic-thermalctld/scripts/thermalctld:661]: Why you get a value from STATE_DB and set it to the platform?

**Review by Junchao-Mellanox — CHANGES_REQUESTED** [review]

**Comment by Junchao-Mellanox** [comment]
I don't quite understand the change here. If you want to allow user to change the temperature threshold, you probably need:
1. A CLI to change the CONFIG_DB 
2. A service/daemon to read the CONFIG_DB changes and do action

STATE_DB is used to reflect the "state" of the switch.

File: sonic-thermalctld/scripts/thermalctld (hunk 2 above)
Comment (d-dashkov) [reply sonic-thermalctld/scripts/thermalctld:662]: Second is new value in string and THRESH_COMP_VAL_LIST[th_type] is current value in float, which I cast to string, so they should be equal.
```
second: <class 'str'>111.0
THRESH_COMP_VAL_LIST[th_type]: <class 'float'>111.0
```

File: sonic-thermalctld/scripts/thermalctld (hunk 3 above)
Comment (d-dashkov) [reply sonic-thermalctld/scripts/thermalctld:664]: This was misunderstood by the SONiC DB architecture and now I have moved the new values to CONFIG_DB as you suggested.

**Pull request closed without merging**